erpnext_shopify/erpnext_shopify/doctype/shopify_settings/shopify_settings.py

```diff
@@ -29,19 +52,23 @@ def get_series():
 def sync_shopify():
 	shopify_settings = frappe.get_doc("Shopify Settings", "Shopify Settings")
 	
-	if not frappe.session.user:
-		user = frappe.db.sql("""select parent from tabUserRole 
-			where role = "System Manager" and parent not in ('administrator', "Administrator") limit 1""", as_list=1)[0][0]
-		frappe.set_user(user)
+	if shopify_settings.enable_shopify:	
+		if not frappe.session.user:
+			frappe.set_user("Administrator")
 		
-	if shopify_settings.enable_shopify:
 		try :
 			sync_products(shopify_settings.price_list, shopify_settings.warehouse)
 			sync_customers()
 			sync_orders()
-			
+		
 		except ShopifyError:
-			pass			
+			shopify_settings.erpnext_shopify = 0
+			shopify_settings.save()
+	else:
+		frappe.msgprint(_("""Invalid acess
```

Comment:
Don't use html tags in msgprint. Also there is a typo. 

The message should only be displayed if the user explicitly clicks on 'Sync Shopify' from UI. Else don't throw exception if via scheduler. 

Use this instead: 

```
elif frappe.local.form_dict.cmd=="{write the cmd here}":
    frappe.throw(_("Shopify connector is not enabled. Click on 'Connect to Shopify' to connect ERPNext and your Shopify store."))
```